### PR TITLE
Remove empty try's

### PR DESCRIPTION
**Pull request by danmosemsft**
Just remove cases where we still had empty try's.

This isn't very interesting, but what is the status of CER's in CoreCLR? What does `RuntimeHelpers.PrepareConstrainedRegions();` do? Is there cleanup possible?

### CODE REVIEW

File: src/mscorlib/shared/System/DateTime.cs [hunk 1]
```diff
@@ -1026,9 +1026,7 @@ public int Month
 
         // Returns a DateTime representing the current date and time. The
         // resolution of the returned value depends on the system timer. For
-        // Windows NT 3.5 and later the timer resolution is approximately 10ms,
-        // for Windows NT 3.1 it is approximately 16ms, and for Windows 95 and 98
-        // it is approximately 55ms.
+        // Windows NT 3.5 and later the timer resolution is approximately 10ms.
```
Comment (mikedn) [inline src/mscorlib/shared/System/DateTime.cs]: And 10ms is out of date - see https://github.com/dotnet/coreclr/pull/9736

Comment (mikedn) [inline src/mscorlib/shared/System/DateTime.cs]: LOL, you fixed it while I was searching for that PR :)

Comment (danmosemsft) [inline src/mscorlib/shared/System/DateTime.cs]: Yes, I dug up exactly that PR :)

**Comment by janvorli** [comment]
@danmosemsft why are we removing the empty trys? Isn't the thread abort protection still needed in those regions? Funceval in VS uses thread abort to abort evaluation of functions that took too long time to execute and the changes you've made would seem to lead to leaks in case the abort happens in the region that were protected by the finally blocks with empty try before.

**Comment by danmosemsft** [comment]
My understanding was thread abort was no longer supported. There was a jit change recently that elided empty trys. Sounds like I misunderstood, or corefx is different?
Cc @jkotas

**Comment by jkotas** [comment]
Thread abort in .NET Core is used for the limited func eval case only. The runtime and framework is not build and tested to be fully reliable in the presence of thread aborts.

**Comment by danmosemsft** [comment]
Was that broken by 119b04c1d3b7a9842eaaacb974e991e5d454f2b4 ?

**Comment by AndyAyersMS** [comment]
From the jit side, finally cloning also has a similar effect -- finally clauses are cloned to streamline the non-exception path performance and the clones are not reported as finallys to the runtime (see #8551).

**Comment by danmosemsft** [comment]
@AndyAyersMS can you clarify, do you mean that 119b04c  means that empty try clauses are never seen by the runtime anyway?

**Comment by AndyAyersMS** [comment]
Not quite, just that empty try removal and finally cloning have similar impact on the way that the code originally within the finally clause is reported to the runtime. 

Finally cloning is more general and has more widespread impact and also came first, so it is the main culprit in the behavior change.

**Comment by danmosemsft** [comment]
OK I'm still confused. Specifically: does the empty try pattern, used in the desktop framework to indicate code that must run on thread abort, similarly functional on .NET Core? If it's not, it sounds like that may surprise @janvorli / @jkotas who expect it to be.

**Comment by AndyAyersMS** [comment]
On .Net Core, when the jit optimizes the code, then protection against thread abort is no longer guaranteed for code in finally clauses. Whether the try is empty or not empty does not matter.

If the jit is not optimizing the code, what happens to code in finally clauses during thread abort is up to the runtime.

**Comment by danmosemsft** [comment]
OK. Given this @janvorli  / @jkotas would you prefer I leave the try's as they are, or remove them ?

**Comment by jkotas** [comment]
I do not see a problem with removing them. There are hundreds of places that are not robust against thread abort. Having a few more won't make a difference. We are not building .NET Core to be fully reliable in the presence of thread abort.

**Pull request merged**